apps/server/src/modules/deletion/controller/api-test/deletion-request-create.api.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { ExecutionContext, INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Request } from 'express';
+import { AuthGuard } from '@nestjs/passport';
+import { EntityManager } from '@mikro-orm/mongodb';
+import { TestXApiKeyClient } from '@shared/testing';
+import { AdminApiServerTestModule } from '../../../server/admin-api.server.module';
+import { DeletionRequestBodyProps, DeletionRequestResponse } from '../dto';
+import { DeletionDomainModel } from '../../domain/types';
+import { DeletionRequestEntity } from '../../entity';
+
+const baseRouteName = '/deletionRequests';
+
+describe(`deletionRequest create (api)`, () => {
+	let app: INestApplication;
+	let em: EntityManager;
+	let testXApiKeyClient: TestXApiKeyClient;
+	const API_KEY = '1ab2c3d4e5f61ab2c3d4e5f6';
+
+	beforeAll(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [AdminApiServerTestModule],
+		})
+			.overrideGuard(AuthGuard('api-key'))
+			.useValue({
+				canActivate(context: ExecutionContext) {
+					const req: Request = context.switchToHttp().getRequest();
+					req.headers['X-API-KEY'] = API_KEY;
+					return true;
+				},
+			})
+			.compile();
+
+		app = module.createNestApplication();
+		await app.init();
+		em = module.get(EntityManager);
+		testXApiKeyClient = new TestXApiKeyClient(app, baseRouteName);
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	const setup = () => {
```

Comment:
Please move it into the when senario description.
If you change the TestApiClient/TestXApiKeyClient than you must also add more when senarios for example api key is missed, or is invalid. It should run into different error results.